csharp_dotnetcore/Using-Cards/Using_Cards/Startup.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.BotFramework;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Linq;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Integration;
+using Microsoft.Extensions.Options;
+
+namespace Using_Cards
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            try
+            {
+                services.AddBot<CardsBot>(options =>
+                {
+                    options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+
+                    // The CatchExceptionMiddleware provides a top-level exception handler for your bot. 
```

Comment:
We don't have a CatchExceptionMiddleware anymore. This comment is obsolete. 

Please follow the Error Handling pattern in DaveTa's samples.

Reply:
Unclear on this, looking into it